kubernetes_state/datadog_checks/kubernetes_state/kubernetes_state.py

```diff
@@ -511,6 +516,32 @@ def kube_job_status_succeeded(self, metric, scraper_config):
                     tags.append(self._format_tag(label_name, label_value, scraper_config))
             self.job_succeeded_count[frozenset(tags)] += sample[self.SAMPLE_VALUE]
 
+    def kube_job_status_start_time(self, metric, scraper_config):
+        for sample in metric.samples:
+            tags = list(scraper_config['custom_tags'])
+            for label_name, label_value in iteritems(sample[self.SAMPLE_LABELS]):
+                tags.append(self._format_tag(label_name, label_value, scraper_config))
+            self.job_start_time[tuple(tags)] = int(sample[self.SAMPLE_VALUE])
+
+    def kube_job_status_completion_time(self, metric, scraper_config):
+        for sample in metric.samples:
+            tags = list(scraper_config['custom_tags'])
+            for label_name, label_value in iteritems(sample[self.SAMPLE_LABELS]):
+                tags.append(self._format_tag(label_name, label_value, scraper_config))
+            self.job_completion_time[tuple(tags)] = int(sample[self.SAMPLE_VALUE])
+
+    def kube_job_status_duration_time(self, scraper_config):
+        metric_name = scraper_config['namespace'] + '.job.duration_time'
+        for job_tags, completion_time in iteritems(self.job_completion_time):
+            start_time = self.job_start_time[job_tags]
```

Comment:
`self.job_start_time[job_tags]`

Can this index lookup fail for any reason ? i.e. `completion time` is collected but the `start time` is missing for a combination of `job_tags`.

If yes, a protection might be needed against IndexError.

Reply:
Since start_time is given at the time of job execution, probably that case does not exist :/
https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/staging/src/k8s.io/api/batch/v1/types.go#L128

Reply:
I think you are right, that case should not exist. I guess if completion time is reported, start time is probably reported too.